hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/TargetWithActionStatus.java

```diff
@@ -31,6 +35,12 @@ public TargetWithActionStatus(final Target target, final Status status) {
         this.target = target;
     }
 
+    public TargetWithActionStatus(final Target target, final Status status, final Optional<Integer> lastActionStatusCode) {
```

Comment:
Would have expected an Integer /int to be passed in directly.
The Optional can then be created in the constructor using Optional.ofNullable(...).